bugsnag-android-core/src/test/java/com/bugsnag/android/JournaledDocumentStressTest.kt

```diff
@@ -0,0 +1,145 @@
+package com.bugsnag.android
+
+import com.bugsnag.android.internal.JournaledDocument
+import com.bugsnag.android.internal.JsonHelper
+import org.junit.Assert
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import java.io.File
+import java.lang.Thread
+
+class JournaledDocumentStressTest {
+    @Rule
+    @JvmField
+    var folder: TemporaryFolder = TemporaryFolder()
+
+    @Test
+    fun testOverflow() {
+        val baseDocumentPath = folder.newFile("mydocument")
+        val bufferSize = 1000L
+        val entryCount = 50000
+        val expectedDocument = mutableMapOf<String, Any>()
+        for (i in 1..entryCount) {
+            expectedDocument["a$i"] = "x"
+        }
+        val document = JournaledDocument(
+            baseDocumentPath,
+            JournaledDocumentTest.standardType,
+            JournaledDocumentTest.standardVersion,
+            bufferSize,
+            bufferSize,
+            mutableMapOf()
+        )
+
+        val modifyThread = Thread {
+            for (i in 1..entryCount) {
+                document.addCommand("a$i", "x")
+            }
+        }
+        modifyThread.start()
+
+        var value: Any? = null
+        var i = 0
+        val maxI = entryCount / 10
+
+        while (modifyThread.state != Thread.State.TERMINATED) {
+            document.forEach {
+                // Just cycle through all entries
+            }
+            value = document["a$i"]
+            i = (i + 1) % maxI
+        }
+
+        // "use" value so that it doesn't get optimized away
+        print(value)
```

Comment:
Can an assertion be added to check each entry is valid? That would avoid the need for printing things to try and stop the compiler optimizing things away.